py-httpcore: add v1.0.5

qwertos opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

alex391 committed:
py-httpcore: Added new version

wdconinc commented on var/spack/repos/builtin/packages/py-httpcore/package.py:
```diff
@@ -15,15 +15,18 @@ class PyHttpcore(PythonPackage):
-    depends_on("py-setuptools", type="build")
-    depends_on("py-h11@0.13:0.14", when="@0.16.3", type=("build", "run"))
+    depends_on("py-setuptools", when="@0", type="build")
+    depends_on("py-hatchling", when="@1.0.5:", type="build")
+    depends_on("py-hatch-fancy-pypi-readme", when="@1.0.5:", type="build")
```
```suggestion
    depends_on("py-hatchling", when="@1:", type="build")
    depends_on("py-hatch-fancy-pypi-readme", when="@1:", type="build")
```

wdconinc commented on var/spack/repos/builtin/packages/py-httpcore/package.py:
```diff
@@ -15,15 +15,18 @@ class PyHttpcore(PythonPackage):
+    depends_on("py-hatchling", when="@1.0.5:", type="build")
+    depends_on("py-hatch-fancy-pypi-readme", when="@1.0.5:", type="build")
+    depends_on("py-h11@0.13:0.14", when="@0.16.3:", type=("build", "run"))
     depends_on("py-h11@0.11:0.12", type=("build", "run"), when="@0.14.7")
     depends_on("py-h11@0.8:0.9", type=("build", "run"), when="@0.11.0")
-    depends_on("py-sniffio@1", type=("build", "run"))
+    depends_on("py-sniffio@1", type=("build", "run"), when="@0")
```
(Optional) It may be better to specify these dependencies with their actual `when` conditions, even if the versions are not included. E.g. `py-h11@0.13:0.14` probably applied before 0.16.3 as well, mabye for `@0.15:` or so. There is no need to

wdconinc reviewed (commented):
Looks good (and hash for 1.0.5 verified). We often benefit from having the last pre-major release version as well, in this case 0.18.0. Would you be able to add that one as well while you are upgrading this package?

qwertos committed:
[py-httpcore]

- added version 0.18.0
- restructured dependencies as everything has a when and
  type/when ordering was all over the place

wdconinc commented on var/spack/repos/builtin/packages/py-httpcore/package.py:
```diff
@@ -15,15 +15,24 @@ class PyHttpcore(PythonPackage):
-    depends_on("py-anyio@3:4", when="@0.16.3", type=("build", "run"))
-    depends_on("py-anyio@3", type=("build", "run"), when="@0.14.7")
-    depends_on("py-certifi", type=("build", "run"), when="@0.14.7:")
+    depends_on("py-setuptools", when="@:1.16.3", type="build")
```
Typo?
```suggestion
    depends_on("py-setuptools", when="@:0.16.3", type="build")
```

wdconinc commented on var/spack/repos/builtin/packages/py-httpcore/package.py:
```diff
@@ -15,15 +15,24 @@ class PyHttpcore(PythonPackage):
+        depends_on("py-anyio@3", when="@0.14.7")
+        depends_on("py-anyio@3:4", when="@0.16.3:0.18")
+
+        depends_on("py-certifi", when="@0.14.7:")
```
Thanks for bringing organization. Can I ask one last thing, and that's moving the certify dependency after the h11 dependency? Those are the active one, and it helps a lot when updating package versions when the active dependencies are separate from the historical dependencies.

qwertos committed:
[py-httpcore] ordered dependencies in the order listed in v1.0.5 pyproject.toml

wdconinc reviewed (approved):
Excellent! 

Note that rebases are not required, and may in fact slow down when a PR gets merged. See https://github.com/spack/spack-infrastructure/blob/main/docs/deferred_pipelines.md for details.
